### DISCUSSION TITLE
Is it possible to provision resources for Cloud Foundry spaces?

### DISCUSSION THREAD

**Discussion by AFK-Python** [SAP/terraform-provider-btp#510] [Q&A] [answered]
After creating a cloud foundry environment instance as so:
```
resource "btp_subaccount_environment_instance" "cloudfoundry" {
  subaccount_id    = btp_subaccount.tfsubaccount.id
  name             = "tf-cf-environment" # <global_account_name>-<subaccount name>
  environment_type = "cloudfoundry"
  service_name     = "cloudfoundry"
  plan_name        = "standard"
  landscape_label  = "cf-us10" # TODO: report that landscape label is required and not option

  # some regions offer multiple environments of a kind and you must explicitly select the target environment in which
  # the instance shall be created. 
  # available environments can be looked up using the btp_subaccount_environments datasource
  parameters = jsonencode({
    instance_name = "my-cf-org-name" # <subaccount-name>-tf
  })
}
```

I see that no spaces are created. Do I have to mix and mash with the cloudfoundry tf provider in order to create a space and the necessary resources for it?

Additionally, would the lack of a created space be responsible for issues with service instances being created? For example, when I try to create a xsuaa instance with this code:
```
resource "btp_subaccount_service_instance" "test-instance" {
  subaccount_id  = btp_subaccount.tfsubaccount.id
  serviceplan_id = "ba8d82ab-2010-4809-8a58-20757e9bf644"
  name           = "xsuaa-instance"
}
```

I get this error:
```
│ Error: API Error Creating Resource Service Instance (Subaccount)
│ 
│   with btp_subaccount_service_instance.test-instance,
│   on main.tf line 142, in resource "btp_subaccount_service_instance" "test-instance":
│  142: resource "btp_subaccount_service_instance" "test-instance" {
│ 
│ BrokerError
```

Or is this perhaps a service plan id error? Do you have a better way of getting the service plan ids instead of the data source? It is a little hard to tell what is what via the data source.

Thanks in advance!

**Comment by lechnerc77** [comment, marked as answer]
Hi @AFK-Python ,

The Terraform provider for SAP BTP is based on the BTP CLI (server) and, hence, its functionality is by design restricted to the BTP-specific resources. 
When it comes to Cloud Foundry (or Kubernetes) specific resources, you must switch to the corresponding provider. To create a CF space, you have to mix in the CF provider.

When it comes to service instances it follows the same pattern: if you want to create a service instance in a CF space you need to make use of the CF provider. If you want to create a service instance that is not specific to a runtime/environment, you can use the Terraform provider for SAP BTP. The later would correspond to the creation of a service instance in the BTP cockpit in the environment "other".

**Reply by AFK-Python** [reply]
I see. Thank you very much for your response! I will look into the CF TF provider.

**Comment by mauriciolauffer** [comment]
Have a look at: [https://registry.terraform.io/providers/cloudfoundry-community/cloudfoundry/latest](https://registry.terraform.io/providers/cloudfoundry-community/cloudfoundry/latest)
